src/cache_filesystem.cpp

```diff
@@ -361,17 +361,32 @@ unique_ptr<FileHandle> CacheFileSystem::GetOrCreateFileHandleForRead(const strin
 	unique_ptr<FileHandle> file_handle = nullptr;
 	{
 		const auto latency_guard = profile_collector->RecordOperationStart(IoOperation::kOpen);
-		file_handle = internal_filesystem->OpenFile(path, flags | FileOpenFlags::FILE_FLAGS_PARALLEL_ACCESS, opener);
+		file_handle = internal_filesystem->OpenFile(file, flags | FileOpenFlags::FILE_FLAGS_PARALLEL_ACCESS, opener);
 	}
 	DUCKDB_LOG_OPEN_CACHE_MISS_PTR((file_handle));
 	return CreateCacheFileHandleForRead(std::move(file_handle));
 }
 
+// LCOV_EXCL_START
+unique_ptr<FileHandle> CacheFileSystem::OpenFileExtended(const OpenFileInfo &file, FileOpenFlags flags,
+                                                         optional_ptr<FileOpener> opener) {
+	InitializeGlobalConfig(opener);
+	if (flags.OpenForReading()) {
```

Comment:
A reminder to myself.

There're two things to followup:
- If we want to enable cache file handle for write, we should at least evict cache on write operations
- Otherwise we simply check `flags.OpenForReading() && !flags.OpenForWriting()`